### PR TITLE
11-get-api-articles-topicquery

### PR DESCRIPTION
**Pull request by klowther15**
completes challenge 11-get-api-articles-topicquery and completes some recommended refactoring

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -257,6 +282,28 @@ describe('GET: api/users', () => {
...
+        return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
+        .then(({ body }) => {
+            expect(body).toHaveLength(1)
+            body.forEach((article) => {
+                expect(article.topic).toBe("cats")
+            });
+        });
+    });
```
Comment (MKD01) [inline __tests__/integration.test.js:296]: Nice first test

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -257,6 +282,28 @@ describe('GET: api/users', () => {
...
+    test('404: gives appropriate error when topic does not exist', () => {
+        return request(app)
+        .get('/api/articles?topic=banana')
+        .expect(404)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Not Found")
+        })
+    })
+});
```
Comment (MKD01) [inline __tests__/integration.test.js:305]: Really good error handling

File: __tests__/integration.test.js [hunk 3]
```diff
@@ -257,6 +282,28 @@ describe('GET: api/users', () => {
...
+
```
Comment (MKD01) [inline __tests__/integration.test.js:306]: One really important test that is missing from this is a test to ensure we receive an empty array when given a topic that exists but has no associated articles. You will have to look at the data to see which topic this will be.

File: endpoints.json [hunk 4]
```diff
@@ -103,6 +104,21 @@
...
+      {
+        "title": "Seafood substitutions are increasing",
+        "topic": "cooking",
+        "author": "weegembump",
+        "body": "Text from the article..",
+        "created_at": "2018-05-30T15:59:13.341Z",
+        "votes": 0,
+        "comment_count": 6
+      }
+    ]
```
Comment (MKD01) [inline endpoints.json:121]: You dont need to include this as your previous entry for /api/articles already has this with the queries array filled with the correct info including topic.

**Review by MKD01 — COMMENTED** [review]
Really good work. There is a missing test and some small changes to be made before you move on from this ticket. Once those changes have been made please merge without needing to send an updated pr
